### DISCUSSION TITLE
Colorize image

### DISCUSSION THREAD

**Discussion by jaka1-newbie** [ImageMagick/ImageMagick#2810] [Help]
convert back.png -fuzz 80% -channel a -fill #F57900 -opaque black +channel back2.png
![back](https://user-images.githubusercontent.com/67605008/98597287-76242b80-22e1-11eb-939f-d2bcfd897ea0.png)

output file in black color((
Fred i need to help

**Comment by fmw42** [comment]
What is it you are trying to do?  What is your IM version and platform?

**Comment by jaka1-newbie** [comment]
im7 win7
try to change color to #F57900

**Reply by fmw42** [reply]
Change color of what? Your command is trying to colorize the alpha channel. The alpha channel can only have 8-bits, not 24 or 32 bits. You have to colorize the image, not the alpha channel.  So what should the output be like? Can you explain?

**Comment by jaka1-newbie** [comment]
I try to change color of my image back.png

**Reply by fmw42** [reply]
Is this what you want?

magick img.png -fill "#F57900" -colorize 100 x.png


![x](https://user-images.githubusercontent.com/13458218/98598511-6da30500-228f-11eb-8df7-a7db7cfbb1cf.png)

**Reply by jaka1-newbie** [reply]
> 
> 
> Is this what you want?
> 
> magick img.png -fill "#F57900" -colorize 100 x.png
> 
> ![x](https://user-images.githubusercontent.com/13458218/98598511-6da30500-228f-11eb-8df7-a7db7cfbb1cf.png)

Do we can do this or not?  
back.png gradient:#026470-#F77251 back2.png

**Reply by fmw42** [reply]
Is this what you want?

```
magick img.png -set option:dims "%wx%h" ^
( -size "%[dims]" -define gradient:direction=southeast gradient:"#026470-#F77251" ) ^
( -clone 0 -alpha extract ) ^
-delete 0 ^
-alpha off -compose copy_opacity -composite ^
x2.png
```

![x2](https://user-images.githubusercontent.com/13458218/98739389-1c157b80-235e-11eb-8c04-36f7d07575b1.png)

**Comment by jaka1-newbie** [comment]
THX ! You are GOD of IM!

**Reply by fmw42** [reply]
Or perhaps you just want?

magick img.png ( -clone 0 -alpha off -fill "#F57900" -colorize 100 ) +swap -flatten x.png


![x](https://user-images.githubusercontent.com/13458218/98599726-5402bd00-2291-11eb-9108-550b93eeac72.png)

**Comment by jaka1-newbie** [comment]
Create grapichs by steps:
1. convert -size 1700x2600 xc:transparent back.png
2. convert back.png  1.png -gravity Southwest -geometry 400x400+0+0 -composite  back.png
3. convert back.png -flip 1.png -gravity SouthWest -geometry 400x400+0+0 -composite  back.png
4. convert back.png -flop 1.png -gravity SouthWest -geometry 400x400+0+0 -composite  back.png
5. convert back.png -flip 1.png -gravity SouthWest -geometry 400x400+0+0 -composite  back.png
6. convert back.png -stroke black -strokewidth 4 -draw "line  50,400 50,2200"    back.png
7. convert back.png -stroke black -strokewidth 4 -draw "line  1650,400 1650,2200"    back.png
8. convert back.png -stroke black -strokewidth 4 -draw "line  400,50 1300,50"    back.png
9. convert back.png -stroke black -strokewidth 4 -draw "line  400,2550 1300,2550"    back.png

May be it possible to do with less steps? 

This is 1.png 
![1](https://user-images.githubusercontent.com/67605008/98682022-16269700-236c-11eb-9073-1e84522e7385.png)

This is result
![back](https://user-images.githubusercontent.com/67605008/98682041-1b83e180-236c-11eb-9246-9522bac96eff.png)

**Comment by snibgo** [comment]
You have a number of commands that form a chain. That is, the output of each command is one input to the next command. So they can easily be merged. For example, merge the first two like this:
```
convert -size 1700x2600 xc:transparent 1.png -gravity Southwest -geometry 400x400+0+0 -composite back.png
```

**Comment by jaka1-newbie** [comment]
Trying to put another image on this
convert back.png 1.png -gravity Center -geometry 900x200+0+0  -composite back.png

![back](https://user-images.githubusercontent.com/67605008/99066598-e4067680-25b1-11eb-8215-0704d582edcc.png)
![Floral Borders - Seperators - 21](https://user-images.githubusercontent.com/67605008/99066643-f41e5600-25b1-11eb-8999-9324e04bbc2d.png)


But it don't want be in Center(

**Reply by jaka1-newbie** [reply]
Solve this: 
composite ( -gravity center -resize 15% -geometry +0+590 1.png ) back.png back.png

**Comment by fmw42** [comment]
I suggest you use convert ... -composite. It is more flexible than composite. See 

https://imagemagick.org/Usage/compose/
https://imagemagick.org/Usage/layers/#convert

**Comment by jaka1-newbie** [comment]
![frame](https://user-images.githubusercontent.com/67605008/99519223-44265f80-299a-11eb-8637-2dc80368e9f1.png)
How to change only black color into "#F57900"?

**Comment by jaka1-newbie** [comment]
frame.png -fuzz 40%  -fill #F57900 -opaque black  frameEND.png

THX!

**Reply by jaka1-newbie** [reply]
![b8382febac08](https://user-images.githubusercontent.com/67605008/99914402-e2e9ed80-2d05-11eb-8ae9-b99335fc8053.jpg)
Hi! 
Can we do text with effects like "France" at this picture?

**Reply by fmw42** [reply]
See https://imagemagick.org/Usage/fonts/

**Reply by jaka1-newbie** [reply]
convert -size 320x100 xc:#F9F29B -font Fegirolst -pointsize 70 -fill #F9F29B   -annotate +5+80 'FRANCE' -gamma 2  +level 0,1000 -white-threshold 999 -morphology Distance Chebyshev:1,1000 -auto-level -shade 135x30 -auto-level +level 10,90% font_chebyshev.jpg

I can't understand how t chNGE color(
not so do same

**Reply by fmw42** [reply]
Here is one way. Save the initial text image as black and white as a mask. Then create a solid image of the desired color. Then use the mask to compose multiply the color image with the final text image.

```
convert -size 320x100 xc:"black" \
-font candice -pointsize 50 -fill "white" \
-gravity center -annotate +0+0 'FRANCE' -write mpr:mask \
-gamma 2 +level 0,1000 -white-threshold 999 \
-morphology Distance Chebyshev:1,1000 \
-auto-level -shade 135x30 -auto-level +level 10,90% \
\( +clone -fill gold -colorize 100 \) \
mpr:mask -compose multiply -composite \
font_text.jpg

```


![font_text](https://user-images.githubusercontent.com/13458218/99916863-9381ce00-2cc1-11eb-8d7d-8d7b101eb462.jpg)

Or you can use the mask to also make the background transparent and save as PNG

```
convert -size 320x100 xc:"black" \
-font candice -pointsize 50 -fill "white" \
-gravity center -annotate +0+0 'FRANCE' -write mpr:mask \
-gamma 2 +level 0,1000 -white-threshold 999 \
-morphology Distance Chebyshev:1,1000 \
-auto-level -shade 135x30 -auto-level +level 10,90% \
\( +clone -fill gold -colorize 100 \) \
mpr:mask -compose multiply -composite -compose over \
mpr:mask -alpha off -compose copy_opacity -composite \
font_text2.png
```


![font_text2](https://user-images.githubusercontent.com/13458218/99916999-90d3a880-2cc2-11eb-89a9-2bbd2aab9641.png)
